service/internal/changedetection/components_test.go

```diff
@@ -0,0 +1,172 @@
+						{
+							Catalog:               "",
+							Name:                  "kubernetes",
+							Reference:             "",
+							ReleaseOperatorDeploy: false,
+							Version:               "1.16.9",
+						},
+						{
+							Catalog:               "",
+							Name:                  "aws-cni",
+							Reference:             "",
+							ReleaseOperatorDeploy: false,
+							Version:               "1.7.0",
+						},
+					},
+				},
+			},
+			targetRelease: releasev1alpha1.Release{
+				Spec: releasev1alpha1.ReleaseSpec{
+					Components: []releasev1alpha1.ReleaseSpecComponent{
+						{
+							Catalog:               "",
+							Name:                  "kubernetes",
+							Reference:             "",
+							ReleaseOperatorDeploy: false,
+							Version:               "1.17.11",
+						},
+						{
+							Catalog:               "",
+							Name:                  "calico",
+							Reference:             "",
+							ReleaseOperatorDeploy: false,
+							Version:               "1.1.1",
+						},
+					},
+				},
+			},
+			result: false,
+		},
+		{
+			name:           "case 2",
+			currentRelease: releasev1alpha1.Release{},
+			targetRelease: releasev1alpha1.Release{
+				Spec: releasev1alpha1.ReleaseSpec{
+					Components: []releasev1alpha1.ReleaseSpecComponent{
+						{
+							Catalog:               "",
+							Name:                  "kubernetes",
+							Reference:             "",
+							ReleaseOperatorDeploy: false,
+							Version:               "1.17.11",
+						},
+						{
+							Catalog:               "",
+							Name:                  "calico",
+							Reference:             "",
+							ReleaseOperatorDeploy: false,
+							Version:               "1.1.1",
+						},
+					},
+				},
+			},
+			result: false,
+		},
+		{
+			name: "case 3",
+			currentRelease: releasev1alpha1.Release{
+				Spec: releasev1alpha1.ReleaseSpec{
+					Components: []releasev1alpha1.ReleaseSpecComponent{
+						{
+							Catalog:               "",
+							Name:                  "kubernetes",
+							Reference:             "",
+							ReleaseOperatorDeploy: false,
+							Version:               "1.16.9",
+						},
+						{
+							Catalog:               "",
+							Name:                  "aws-cni",
+							Reference:             "",
+							ReleaseOperatorDeploy: false,
+							Version:               "1.7.0",
+						},
+					},
+				},
+			},
+			targetRelease: releasev1alpha1.Release{
+				Spec: releasev1alpha1.ReleaseSpec{
+					Components: []releasev1alpha1.ReleaseSpecComponent{
+						{
+							Catalog:               "",
+							Name:                  "kubernetes",
+							Reference:             "",
+							ReleaseOperatorDeploy: false,
+							Version:               "1.16.9",
+						},
+						{
+							Catalog:               "",
+							Name:                  "calico",
+							Reference:             "",
+							ReleaseOperatorDeploy: false,
+							Version:               "1.7.0",
+						},
+					},
+				},
+			},
+			result: true,
+		},
```

Comment:
Let's have one production relevant test case where all versions are equal except one so that we know the update should be triggered.

Reply:
Added two more tests, which have all components included. One which should trigger and one which doesn't trigger CF update.